### PR TITLE
feat(ui5-step-input): add input event

### PR DESCRIPTION
**Pull request by tsanislavgatev**
fixes: https://github.com/SAP/ui5-webcomponents/issues/5177

### CODE REVIEW

**Review by unazko — APPROVED** [review]
Works as expected.

File: packages/main/src/StepInput.ts [hunk 1]
```diff
@@ -106,6 +106,16 @@ type StepInputValueStateChangeEventDetail = {
 @event("change", {
 	bubbles: true,
 })
+/**
+ * Fired when the value of the component changes at each keystroke,
+ * and when a suggestion item has been selected.
```
Comment (vladitasev) [inline packages/main/src/StepInput.ts:111]: This is not relevant for step input

File: packages/main/src/StepInput.ts [hunk 2]
```diff
@@ -106,6 +106,16 @@ type StepInputValueStateChangeEventDetail = {
...
+ * @public
+ * @since 2.5.0
```
Comment (vladitasev) [inline packages/main/src/StepInput.ts:113]: this becomes 2.6.0 now

File: packages/main/src/StepInput.ts [hunk 3]
```diff
@@ -379,6 +390,14 @@ class StepInput extends UI5Element implements IFormInputElement {
 		}, 0);
 	}
 
+	_onInput(e: InputEvent) {
```
Comment (vladitasev) [inline packages/main/src/StepInput.ts:393]: `InputEvent` is wrong here - this is the event type that will be fired by the native input (so in Input.ts it's normal that the event listener of this type) - here you need the correct `InputEventDeatil` from the `Input.ts` event types.

File: packages/main/src/StepInput.ts [hunk 4]
```diff
@@ -122,6 +132,7 @@ type StepInputValueStateChangeEventDetail = {
 class StepInput extends UI5Element implements IFormInputElement {
 	eventDetails!: {
 		change: void
+		input: void
```
Comment (vladitasev) [inline packages/main/src/StepInput.ts:135]: I think the `inputType` event detail that exists on `InputEventDetail` is relevant for the user, so it must also be proxied.

File: packages/main/src/StepInput.ts [hunk 5]
```diff
@@ -379,6 +390,14 @@ class StepInput extends UI5Element implements IFormInputElement {
 		}, 0);
 	}
 
+	_onInput(e: InputEvent) {
+		const prevented = !this.fireDecoratorEvent("input");
```
Comment (vladitasev) [inline packages/main/src/StepInput.ts:394]: Fire with parameters (see my comment above in `eventDetails`

**Review by vladitasev — CHANGES_REQUESTED** [review]

**Review by vladitasev — APPROVED** [review]

**Pull request merged**